sdk/core/Azure.Core/src/Messaging/CloudEvent.cs

```diff
@@ -62,11 +69,11 @@ public CloudEvent(string source, string type, BinaryData? data, string? dataCont
         /// Gets or sets an identifier for the event. The combination of <see cref="Id"/> and <see cref="Source"/> must be unique for each distinct event.
         /// If not explicitly set, this will default to a <see cref="Guid"/>.
         /// </summary>
-        public string? Id
+        public string Id
         {
             get
             {
-                return _id;
+                return _id!;
```

Comment:
I think we decided on throwing, no?

Reply:
I think the decision was to say that this could be null, if using skipValidation. There is no reason to throw if the customer wants to get the Id though.

Reply:
/cc @KrzysztofCwalina